modules/http/http.c

```diff
@@ -0,0 +1,331 @@
+/*
+ * Copyright (c) 2016 Marc Falzon
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 as published
+ * by the Free Software Foundation, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * As an additional exemption you are allowed to compile & link against the
+ * OpenSSL libraries as published by the OpenSSL project. See the file
+ * COPYING for details.
+ *
+ */
+
+#include <curl/curl.h>
+
+#include "syslog-names.h"
+#include "http-plugin.h"
+
+static gchar *
+_format_persist_name(LogThrDestDriver *s)
+{
+  static gchar persist_name[1024];
+
+  HTTPDestinationDriver *self = (HTTPDestinationDriver *) s;
+
+  g_snprintf(persist_name, sizeof(persist_name), "http(%s,)", self->url);
+
+  return persist_name;
+}
+
+static gchar *
+_format_stats_instance(LogThrDestDriver *s)
+{
+  static gchar stats[1024];
+
+  HTTPDestinationDriver *self = (HTTPDestinationDriver *) s;
+
+  g_snprintf(stats, sizeof(stats), "http,%s", self->url);
+
+  return stats;
+}
+
+static size_t
+_http_write_cb(char *ptr, size_t size, size_t nmemb, void *userdata)
+{
+    // Discard response content
+    return nmemb * size;
+}
+
+static void
+_thread_init(LogThrDestDriver *s)
+{
+  HTTPDestinationDriver *self = (HTTPDestinationDriver *) s;
+
+  curl_version_info_data *curl_info = curl_version_info(CURLVERSION_NOW);
+  if (!self->user_agent)
+    self->user_agent = g_strdup_printf("syslog-ng %s/libcurl %s",
+                                        SYSLOG_NG_VERSION, curl_info->version);
+}
+
+static void
+_thread_deinit(LogThrDestDriver *s)
+{
+}
+
+static gboolean
+_connect(LogThrDestDriver *s)
+{
+  return TRUE;
+}
+
+static void
+_disconnect(LogThrDestDriver *s)
+{
+}
+
+static struct curl_slist *
+_get_curl_headers(HTTPDestinationDriver *self, LogMessage *msg)
+{
+  GList *header = NULL;
+  struct curl_slist *curl_headers = NULL;
+  gchar header_host[128] = {0};
+  gchar header_program[32] = {0};
+  gchar header_facility[32] = {0};
+  gchar header_level[32] = {0};
+
+  g_snprintf(header_host, sizeof(header_host),
+    "X-Syslog-Host: %s", log_msg_get_value(msg, LM_V_HOST, NULL));
```

Comment:
@ihrwein  Thanks for pointing it out.
I think the prefix is "X-Syslog-" instead of  just "X-" .  I think it is enough to distinguish itself from other standard parameters.   I can't come up with any scenarios that will cause problem so far.